cmake/scripts/run_test.cmake

```diff
@@ -53,7 +53,8 @@ IF("${EXPECT}" STREQUAL "")
 ENDIF()
 
 EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND}
-  --build ${BINARY_DIR} --target ${TRGT}
+  --build . --target ${TRGT}
```

Comment:
You needed this because you relocated the tests out of ```${BINARY_DIR}```?

If so, I suggest to add a runtime output directory to ```DEAL_II_ADD_TEST``` that puts the executable inside ```${BUILDDIR}``` (instead of ```${BUILDDIR}/Debug```, etc.).